app_build_suite/build_steps/helm.py

```diff
@@ -382,6 +403,72 @@ def pre_run(self, config: argparse.Namespace) -> None:
                 ):
                     raise ValidationError(self.name, f"Value of '{option}' is not a correct boolean.")
 
+    def additional_annotations(self, chart_file_name: str, catalog_name: str) -> Dict[str, Any]:
+        catalog_base_url = f"{self._value_annotation_catalog_base_url}{catalog_name}/{chart_file_name}-meta/"
+        annotations = {}
+        # TODO check if these exist
+        annotations[self._key_annotation_metadata_url] = f"{catalog_base_url}main.yaml"
+        annotations[self._key_annotation_readme_url] = f"{catalog_base_url}README.md"
+        return annotations
+
+    def run(self, config: argparse.Namespace, context: Dict[str, Any]) -> None:
+        if not config.generate_metadata:
+            logger.info("Metadata generation is disabled using 'generate-metadata' option.")
+            return
+        # read current Chart.yaml
+        chart_yaml_path = os.path.join(config.chart_dir, _chart_yaml)
+        with open(chart_yaml_path, "r") as file:
+            chart_yaml = yaml.safe_load(file)
+        # try to guess the package file name. we need it for url generation in annotations
+        chart_name = os.path.basename(config.chart_dir)
+        context[context_key_chart_file_name] = f"{chart_name}-{context[context_key_git_version]}.tgz"
+        # put in generated annotations
+        chart_yaml[self._key_annotations] = {
+            **chart_yaml.get(self._key_annotations, {}),
+            **self.additional_annotations(context[context_key_chart_file_name], config.catalog_name),
+        }
+        # save Chart.yaml
+        save_yaml(chart_yaml_path, chart_yaml)
```

Comment:
I think we should take backups of these files first. `GitVersionSetter` does that already here: https://github.com/giantswarm/app-build-suite/blob/01c4853803c362475709d9b85a6d99f73ea56e49/app_build_suite/build_steps/helm.py#L144

You can probably set a flag in `context` to mark if the backups were already taken and skip it if they were. Otherwise, probably we should extract a small shared `take_chart_backup()` method here and run it. 

I'm also thinking if we should restore original files in post-build (configurable). It's annoying when a build tool changes your files and you don't want to commit the changes.